pkg/scalers/scaler.go

```diff
@@ -110,6 +111,21 @@ func GenerateMetricNameWithIndex(scalerIndex int, metricName string) string {
 	return fmt.Sprintf("s%d-%s", scalerIndex, metricName)
 }
 
+// GenerateMetricNameWithoutIndex removes the index prefix from the metric name
+func GenerateMetricNameWithoutIndex(scalerIndex int, metricName string) (string, error) {
```

Comment:
I'd call this function something like `RemoveIndexFromMetricName`. It's more clear IMO.
WDYT?

Reply:
Was my initial choice, then I decided to keep the naming similar. Will revert.

Reply:
Corrected.